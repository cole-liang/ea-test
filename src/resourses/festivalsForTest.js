export const sourceList = [
  {
    name: "LOL-palooza",
    bands: [
      {
        name: "Winter Primates",
        recordLabel: ""
      },
      {
        name: "Squint-281",
        recordLabel: "Outerscope"
      },
      {
        name: "Auditones",
        recordLabel: "ACR"
      }
    ]
  },
  {
    name: "Twisted Tour",
    bands: [
      {
        name: "Squint-281"
      }
    ]
  },
  {
    bands: [
      {
        name: "Critter Girls",
        recordLabel: "ACR"
      },
      {
        name: "Propeller",
        recordLabel: "Pacific Records"
      }
    ]
  }
];

export const expectedList = [
  {
    recordLabel: "ACR",
    bands: [
      { bandName: "Auditones", festivals: ["LOL-palooza"] },
      { bandName: "Critter Girls", festivals: [] }
    ]
  },
  {
    recordLabel: "Outerscope",
    bands: [
      { bandName: "Squint-281", festivals: ["LOL-palooza", "Twisted Tour"] }
    ]
  },
  {
    recordLabel: "Pacific Records",
    bands: [{ bandName: "Propeller", festivals: [] }]
  },
  {
    recordLabel: "No Specified Record",
    bands: [{ bandName: "Winter Primates", festivals: ["LOL-palooza"] }]
  }
];

export const flattenSourceList = [
  {
    name: "LOL-palooza",
    bands: [
      {
        name: "Winter Primates",
        recordLabel: ""
      },
      {
        name: "Squint-281",
        recordLabel: "Outerscope"
      }
    ]
  },
  {
    name: "Twisted Tour",
    bands: [
      {
        name: "Squint-281"
      }
    ]
  },
  {
    bands: [
      {
        name: "Critter Girls",
        recordLabel: "ACR"
      }
    ]
  }
];

export const flattenExpectedList = [
  {
    festivalName: "LOL-palooza",
    name: "Winter Primates",
    recordLabel: ""
  },
  {
    festivalName: "LOL-palooza",
    name: "Squint-281",
    recordLabel: "Outerscope"
  },
  {
    festivalName: "Twisted Tour",
    name: "Squint-281"
  },
  {
    festivalName: undefined,
    name: "Critter Girls",
    recordLabel: "ACR"
  }
];

export const banFesExpectedObj = {
  "Winter Primates": ["LOL-palooza"],
  "Squint-281": ["LOL-palooza", "Twisted Tour"],
  "Critter Girls": []
};

export const recBanExpectedObj = {
  ACR: ["Critter Girls"],
  "": ["Winter Primates"],
  Outerscope: ["Squint-281"],
  undefined: ["Squint-281"]
};

export const expectedBandOcc = new Set(["Squint-281", "Critter Girls"]);

export const expectedFinalResult = [
  {
    recordLabel: "ACR",
    bands: [{ bandName: "Critter Girls", festivals: [] }]
  },
  {
    recordLabel: "Outerscope",
    bands: [
      { bandName: "Squint-281", festivals: ["LOL-palooza", "Twisted Tour"] }
    ]
  },
  {
    recordLabel: "No Specified Record",
    bands: [{ bandName: "Winter Primates", festivals: ["LOL-palooza"] }]
  }
];
