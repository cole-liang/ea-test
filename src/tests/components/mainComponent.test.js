import React from "react";
import { shallow } from "enzyme";
import MainComponent from "./../../components/mainComponent";
import * as data from "../../resourses/festivalsForTest";

describe("MainComponent component test", () => {
  const wrapper = shallow(<MainComponent></MainComponent>);

  it("should render MainComponent", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render MainComponent without receiving data", () => {
    wrapper.setState({ festivals: [] });
    expect(wrapper).toMatchSnapshot();
  });

  describe("Format data", () => {
    const instance = wrapper.instance();

    it("should format data from API", () => {
      const input = data.sourceList;
      const expected = data.expectedList;

      const result = instance.transformFormat(input);

      expect(result).toStrictEqual(expected);
    });

    it("should flattern Data", () => {
      const input = data.flattenSourceList;
      const expected = data.flattenExpectedList;

      const result = instance.flattenData(input);

      expect(result).toStrictEqual(expected);
    });

    it("should generate band_festival object", () => {
      const input = data.flattenExpectedList;
      const expected = data.banFesExpectedObj;

      const result = instance.generateBandFestivalObj(input);

      expect(result).toStrictEqual(expected);
    });

    it("should generate recordLabel_band object and bandOccurrence set", () => {
      const input = data.flattenExpectedList;
      const expected = {
        recordLabel_bands: data.recBanExpectedObj,
        bandOccurance: data.expectedBandOcc
      };

      const result = instance.generateRecordBandObjAndBandOcc(input);

      expect(result).toStrictEqual(expected);
    });

    it("should generate final result", () => {
      const { recBanExpectedObj, banFesExpectedObj, expectedBandOcc } = data;
      const expected = data.expectedFinalResult;

      const result = instance.generateFormattedResult(
        recBanExpectedObj,
        banFesExpectedObj,
        expectedBandOcc
      );

      expect(result).toStrictEqual(expected);
    });
  });
});
