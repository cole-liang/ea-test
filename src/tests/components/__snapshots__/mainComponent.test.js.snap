// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent component test should render MainComponent 1`] = `
<div>
  Cannot load data, please refresh the page later
</div>
`;

exports[`MainComponent component test should render MainComponent without receiving data 1`] = `
<div>
  Cannot load data, please refresh the page later
</div>
`;
